packages/typescript-estree/src/create-program/createProjectService.ts

```diff
@@ -0,0 +1,70 @@
+import * as ts from 'typescript';
+import * as tsserver from 'typescript/lib/tsserverlibrary';
+
+export function createProjectService() {
+  const compilerOptions = { /* todo */ strict: true };
+  const compilerHost = tsserver.createCompilerHost(compilerOptions, true);
+
+  // TODO: see getWatchProgramsForProjects
+  // We don't watch the disk, we just refer to these when ESLint calls us
+  // there's a whole separate update pass in maybeInvalidateProgram at the bottom of getWatchProgramsForProjects
+  // (this "goes nuclear on TypeScript")
+  const watchFile = (
+    path: string,
+    callback: ts.FileWatcherCallback,
+  ): ts.FileWatcher => {
+    // todo (or just ... stub out?)
+    return { close() {} };
+  };
+
+  const watchDirectory = (
+    path: string,
+    callback: ts.DirectoryWatcherCallback,
+  ) => {
+    // todo (or just ... stub out?)
+    return { close() {} };
+  };
+
+  const system = {
+    ...ts.sys,
+    watchFile,
+    watchDirectory,
+    setTimeout,
+    clearTimeout,
+    setImmediate,
+    clearImmediate,
+  };
+
+  const projectService = new tsserver.server.ProjectService({
+    host: system,
+
+    // todo: look into these
+    cancellationToken: { isCancellationRequested: () => false },
+    useSingleInferredProject: false, // ???
+    useInferredProjectPerProjectRoot: false, // ???
+    typingsInstaller: {
+      attach: () => {},
+      enqueueInstallTypingsRequest: () => {},
+      installPackage: async (): Promise<never> => {
+        throw new Error('pls no');
+      },
+      isKnownTypesPackageName: () => false,
+      onProjectClosed: () => {},
+      globalTypingsCacheLocation: '',
+    },
+    logger: {
+      close() {},
+      hasLevel: () => false,
+      loggingEnabled: () => false,
+      perftrc() {},
+      info() {},
+      startGroup() {},
+      endGroup() {},
+      msg() {},
+      getLogFileName: () => undefined,
+    },
+    session: undefined,
+  });
```

Comment:
I feel like some of these could be optional, though I think we may instead want to acutally provide ATA because that's a thing that is useful for people writing javascript. Otherwise, type-using rules will not actually actually get types.

Reply:
As for defaults, I think what you've written is acutally fine.

Reply:
> provide ATA because that's a thing that is useful for people writing javascript

AH, we've been bit in the past by trying to do things for users. Even ignoring performance penalties (e.g. #5857's explanation), folks find it confusing for types to be different in different tools.